lib/VerticalDelegate.js

```diff
@@ -408,7 +408,8 @@ module.exports = {
 		if (!list.hasReset) return this.reset(list);
 
 		var cpp = this.controlsPerPage(list),
-			end = Math.max(list.$.page1.start, list.$.page2.start) + cpp;
+			end = Math.max(list.$.page1.start, list.$.page2.start) + cpp - 1,
+			idx = props.index - 1;
```

Comment:
the pros.index means previous model count. so if the previous model count is 38 it will have 2 pages with 19 models each and ideally should return 1 as last page index. but if we pass 38 which divided by 19 will give 2. this is I meant by boundary condition. if pros.index is 0-18 -> page 0, 19-37-> page 1 38-56 -> page 3 and so on.